### PR TITLE
Updates to address fips-setup and inter dependency modules #189

### PR DESCRIPTION
**Pull request by prb112**
Signed-off-by: Paul Bastide <pbastide@us.ibm.com>

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -144,6 +144,11 @@ module "fips" {
     ibm = ibm.ibm-cloud-powervs
   }
 
+  depends_on = [
+    module.bastion,
+    module.nbde
+  ]
+
```
Comment (prb112) [inline main.tf:151]: Ensures FIPS isn't executed when nbde/bastion are setup.

File: modules/1_bastion/outputs.tf [hunk 2]
```diff
@@ -30,7 +30,7 @@ output "bastion_public_ip" {
 
 output "bastion_instance_ids" {
   depends_on = [null_resource.bastion_packages]
-  value      = split("/", ibm_pi_instance.bastion[0].id)[1]
+  value      = ibm_pi_instance.bastion.*.id
```
Comment (prb112) [inline modules/1_bastion/outputs.tf:33]: Changed how the input is passed so it remains an array vs string

File: modules/3_fips/variables.tf [hunk 3]
```diff
@@ -52,12 +52,7 @@ variable "bastion_instance_ids" {
     inst_ids = ""
   }
 }
-variable "bastion_public_ip" {
-  default = {
-    count = 1
-    ip    = ""
-  }
-}
+variable "bastion_public_ip" { type = string }
```
Comment (prb112) [inline modules/3_fips/variables.tf:55]: Update so it's a string

File: modules/3_fips/fips.tf [hunk 4]
```diff
@@ -48,7 +48,7 @@ EOF
   }
 
   provisioner "file" {
-    source      = "${path.cwd}/files/fips.yml"
+    source      = "${path.cwd}/modules/3_fips/files/fips.yml"
```
Comment (prb112) [inline modules/3_fips/fips.tf:51]: Update path to fips.yml to include the module path

File: modules/3_fips/fips.tf [hunk 5]
```diff
@@ -48,15 +48,15 @@ EOF
...
     destination = "fips/fips.yml"
   }
 
   provisioner "remote-exec" {
     inline = [
       <<EOF
 echo "Hosts: ${var.tang_ips}"
-echo "[vmhost],${var.tang_ips}" | tr "," "\n\t" > inventory
+echo "[vmhost],${var.tang_ips}" | tr "," "\n\t" > fips/inventory
```
Comment (prb112) [inline modules/3_fips/fips.tf:59]: Changed the inventory path

Comment (gauravpbankar) [reply modules/3_fips/fips.tf:59]: lgtm

File: modules/3_fips/variables.tf (hunk 3 above)
Comment (gauravpbankar) [reply modules/3_fips/variables.tf:55]: lgtm

**Review by gauravpbankar — APPROVED** [review]
Tested the changes and got expected results

File: modules/3_fips/fips.tf (hunk 5 above)
Comment (aditijadhav38) [reply modules/3_fips/fips.tf:59]: LGTM

File: modules/3_fips/variables.tf (hunk 3 above)
Comment (aditijadhav38) [reply modules/3_fips/variables.tf:55]: LGTM

**Comment by prb112** [comment]
Thank you

**Pull request merged**